### PR TITLE
Fix and unify sort orders in default search providers

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below
| Type?             | refacto
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See below
| Fixed ticket?     | -
| Related PRs       | -
| Sponsor company   | -

### What this is about
- Default search providers are used, if no filtering module says that he wants to filter products for current query.
- Every search provider must return sort orders available back to the controller. There was some mess in it.

### Changes
- Removed a hardcode override of sort order in `BestSalesProductSearchProvider`. Added a fallback instead for random order, which is coming from `ps_bestsellers` module. This order is not supported in `getBestSales` function and should be removed from the module - https://github.com/PrestaShop/ps_bestsellers/pull/27.
- Added PHPDoc for all attributes.
- Used `SortOrdersCollection` instead of deprecated `SortOrderFactory`.
- `Relevance/Position in category` order removed from the default set, because it makes sense only in category controller.
- Instead of repeating the same code again and again on some providers, providers now use `getDefaults` of `SortOrdersCollection` to get the basic set - Name desc/asc, price desc/asc. Then we only enrich it by special orders for specific pages:
  - Position order added on category page and search page.
  - Sales order added on best sales page.
  - Date order added on new products page.

### How to test
- ❗Uninstall or disable faceted search.
- Check all FO listing pages and see that sort orders show properly.
- Only difference you should see that "Relevance" or "Position" order won't be available on pages other than category and search. It does not make sense in other pages.

### CODE REVIEW

File: src/Adapter/BestSales/BestSalesProductSearchProvider.php [hunk 1]
```diff
@@ -42,10 +42,16 @@ class BestSalesProductSearchProvider implements ProductSearchProviderInterface
...
+    /**
+     * @var SortOrderFactory
+     */
+    private $sortOrderFactory;
+
     public function __construct(
         TranslatorInterface $translator
     ) {
         $this->translator = $translator;
+        $this->sortOrderFactory = new SortOrderFactory($this->translator);
```
Comment (FabienPapet) [inline src/Adapter/BestSales/BestSalesProductSearchProvider.php]: This one is deprecated, use \PrestaShop\PrestaShop\Core\Product\Search\SortOrdersCollection

Comment (Hlavtox) [reply src/Adapter/BestSales/BestSalesProductSearchProvider.php]: @FabienPapet Fixed, thanks dude!

**Review by kpodemski — DISMISSED** [review]

**Review by SharakPL — DISMISSED** [review]

**Review by SharakPL — DISMISSED** [review]

**Review by SharakPL — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @Hlavtox ,

We have checked your PR it is good !
Automated tests are OK.

Thank you !

**Comment by nicosomb** [comment]
Thank you @Hlavtox and @MhiriFaten !

**Pull request merged**